src/widget/woverview.h

```diff
@@ -32,6 +32,13 @@ class WOverview : public WWidget, public TrackDropTarget {
     void setup(const QDomNode& node, const SkinContext& context);
     virtual void initWithTrack(TrackPointer pTrack);
 
+    enum Type {
```

Comment:
pretty sure Q_ENUMs work on `enum class`es. Sorry, should've specified that we should still keep the enum class.

Reply:
That would be an `enum class` inside a  `class`?
Tried that but pparently it's not accessible elsewhere even tough it's public.

Reply:
_ping_ I need some assistance here.
What's the benefit of having the enum (class) inside the WOverview class? We need it at least in one other unit (prefs), and according to my test that doesn't work unfortunately.

Or, the other way around: what would be the downside of it being a enum class declared in woverview.h but **not** in the WOverview class? With the distinct name `OverviewType` there's no danger of global name clashes IMO, if that's what the `WOverview` prefix  is supposed to prevent.

Reply:
> ping I need some assistance here.

Sorry. I overlooked your previous comment.

> What's the benefit of having the enum (class) inside the WOverview class? 

unscoped enums (plain `enum`) always implicitly convert to their underlying type. In the majority of cases (including here IMO), that is undesired. 

> We need it at least in one other unit (prefs), and according to my test that doesn't work unfortunately.

[I can't reproduce](
https://compiler-explorer.com/#g:!((g:!((g:!((h:codeEditor,i:(filename:'1',fontScale:14,fontUsePx:'0',j:1,lang:c%2B%2B,selection:(endColumn:106,endLineNumber:9,positionColumn:106,positionLineNumber:9,selectionStartColumn:106,selectionStartLineNumber:9,startColumn:106,startLineNumber:9),source:'class+A+%7B%0A++++public:%0A++++//+declare+in+public+section%0A++++enum+class+Enum+%7B%0A++++++++member1,%0A++++++++member2%0A++++%7D%3B%0A++++//+using+enum+to+%22inline%22+enum+members+into+class%0A++++//+needs+C%2B%2B20%3B+only+benefit+is+having+to+type+a+little+less+because+you+can+omit+%60::Type%60+(see+below)%0A++++using+enum+Enum%3B%0A%7D%3B%0A%0Aclass+B+%7B%0A++++//+use+value%0A++++B()+:+member1(A::Enum::member1)+%7B%7D%0A++++//+use+type%0A++++A::Enum+member1%3B%0A++++//+use+%22inlined%22+member%0A++++A::Enum+member2+%3D+A::member2%3B%0A%7D%3B'),l:'5',n:'1',o:'C%2B%2B+source+%231',t:'0')),k:33.333333333333336,l:'4',n:'0',o:'',s:0,t:'0'),(g:!((h:compiler,i:(compiler:g141,filters:(b:'0',binary:'1',binaryObject:'1',commentOnly:'0',debugCalls:'1',demangle:'0',directives:'0',execute:'1',intel:'0',libraryCode:'0',trim:'1',verboseDemangling:'0'),flagsViewOpen:'1',fontScale:14,fontUsePx:'0',j:1,lang:c%2B%2B,libs:!(),options:'-Wall+-Wextra+--std%3Dc%2B%2B20',overrides:!(),selection:(endColumn:1,endLineNumber:1,positionColumn:1,positionLineNumber:1,selectionStartColumn:1,selectionStartLineNumber:1,startColumn:1,startLineNumber:1),source:1),l:'5',n:'0',o:'+x86-64+gcc+14.1+(Editor+%231)',t:'0')),k:33.333333333333336,l:'4',n:'0',o:'',s:0,t:'0'),(g:!((h:output,i:(compilerName:'x86-64+gcc+14.1',editorid:1,fontScale:14,fontUsePx:'0',j:1,wrap:'1'),l:'5',n:'0',o:'Output+of+x86-64+gcc+14.1+(Compiler+%231)',t:'0')),k:33.33333333333333,l:'4',n:'0',o:'',s:0,t:'0')),l:'2',n:'0',o:'',t:'0')),version:4). What code did you test? 

> what would be the downside of it being a enum class declared in woverview.h but not in the WOverview class?

you would have to put it into a namespace that then gets annotated with `Q_NAMESPACE` and make it a Q_ENUM_NS (works, but a little more complicated than just putting it right into the QObject when its already there).

> With the distinct name OverviewType there's no danger of global name clashes IMO, if that's what the WOverview prefix is supposed to prevent.

I'm not too worried about nameclashes. Its more a technical limitation because in order to make an enum a Q_ENUM, Qt's MOC needs to know about it. So it either needs to be inclosed in a Q_NAMESPACE or in a QObject. Since Q_NAMESPACEs work slightly differently from real c++ namespaces, leveraging the QObject is probably less confusing.

Reply:
Ah yeah. I guess thats an explicitly choice from Qt for some reason (mby legacy, idk). The fromValue is fine then IMO. Sorry for dragging this discussion out again. Should've done more research beforehand.

Reply:
Never mind..